cms/djangoapps/contentstore/rest_api/v1/views/videos.py

```diff
@@ -180,3 +181,38 @@ def get(self, request: Request, course_id: str, edx_video_id: str):
         usage_locations = get_video_usage_path(course_key, edx_video_id)
         serializer = VideoUsageSerializer(usage_locations)
         return Response(serializer.data)
+
+
+@view_auth_classes(is_authenticated=True)
+class VideoDownloadView(DeveloperErrorViewMixin, APIView):
+    """
+    View for course video downloads.
+    """
+    @apidocs.schema(
+        parameters=[
+            apidocs.string_parameter("course_id", apidocs.ParameterLocation.PATH, description="Course ID"),
+        ],
+        responses={
+            200: "In case of success, a 200.",
+            401: "The requester is not authenticated",
+            403: "The requester cannot access the specified course",
+            404: "The requested course does not exist",
+        },
+    )
+    @verify_course_exists()
+    def put(self, request: Request, course_id: str) -> Response:
+        """
+        Get an object containing course videos.
+        **Example Request**
+            PUT /api/contentstore/v1/videos/download
```

Comment:
A comment here that the request's body specifies the names of all files to be downloaded would probably be helpful